pkg/server/message.go

```diff
@@ -219,26 +219,26 @@ func (t *TranslateServiceServer) UpdateMessages(
 		return nil, status.Errorf(codes.InvalidArgument, err.Error())
 	}
 
-	all, err := t.repo.LoadMessages(ctx, params.serviceID, repo.LoadMessagesOpts{})
+	all, err := t.repo.LoadTranslation(ctx, params.serviceID, repo.LoadTranslationOpts{})
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "")
 	}
 
-	if !all.HasLanguage(params.messages.Language) {
-		return nil, status.Errorf(codes.NotFound, "no messages for language: '%s'", params.messages.Language)
+	if !all.HasLanguage(params.translation.Language) {
+		return nil, status.Errorf(codes.NotFound, "no translation for language: '%s'", params.translation.Language)
 	}
 
 	// Case for when not original, or uploading original for the first time.
-	updatedMessages := model.MessagesSlice{*params.messages}
+	updatedTranslations := model.TranslationSlice{*params.translation}
 
-	if origIdx := all.OriginalIndex(); params.messages.Original && origIdx != -1 {
+	if origIdx := all.OriginalIndex(); params.translation.Original && origIdx != -1 {
 		oldOriginal := all[origIdx]
 
-		// Mark new or altered original messages as untranslated for all translations.
-		all.MarkUntranslated(oldOriginal.FindChangedMessageIDs(params.messages))
-		// Replace original messages with new ones.
-		all.Replace(*params.messages)
-		// Add missing messages for all translations.
+		// Mark new or altered original translation as untranslated for all translations.
```

Comment:
That comment now doesn't make sense :)
```suggestion
// Compare repo and request original translation. Change status for new or altered translation.messages to UNTRANSLATED for all languages
```